### PR TITLE
Add localization to data table views

### PR DESCRIPTION
**Pull request by wilkie**
**What**: Localizes the raw strings within the data table.

**Why**: To allow other locales and translations for the data table.

**Note**: Also localizes "Clear Table" since it was related, even though it is not explicitly mentioned in the ticket. Also "Actions" and the various import/export, confirmation dialogs, etc... some of which already have localization.

Very simple changes to add localization keys for the strings used in the table into the common i18n JavaScript file. Though, dozens of simple changes!! Sorry for that!!

Reusing localization for: `Add`, `Cancel`, `Yes`, `Delete`, `Deleting...`, `Clear Table`, 'Edit', 'Saving...', 'Save', `Actions`, and some confirmation text

Adding localization for: `addingToTable`, `addPairToTable`, `addRowToTable`, `backToData`, `clearTable`, `confirmImportOverwriteTitle`, `dataTableConvertToString`, `dataTableConvertToNumber`, `dataTableConvertToBoolean`, `dataTableDebugView`, `dataTableTableView`, `dataTableKey`, `dataTableValue`, `deleteTable`, `deleteTableConfirm`, `enterText`, `exportToCSV`, and `importCSV`

## Links

- jira ticket: [FND-2070](https://codedotorg.atlassian.net/jira/software/c/projects/FND/boards/62?modal=detail&selectedIssue=FND-2070)

## Testing story

Manual testing to see the localization strings being used on a development server.

This view now has localized strings for "Add" and "Delete" which already exist! (reuses `add`, `delete`, and `actions`)

image

The delete confirmation dialog now has localized strings: (adds `deleteTable` and `deleteTableConfirm`, reuses `delete`, `cancel`)

image

The data table view is localized: (adds `clearTable`, `importCSV`, `exportToCSV`, `addRowToTable`, `backToData`, `dataTableDebugView`, `dataTableTableView`)

image

The pending text is localized when adding a row: (adds `addingToTable` to match name convention with `addingToProjects`)

image

The column edit dropdown is localized: (adds `convertToString`, `convertToNumber`, `convertToBoolean`; reuses `delete` and `rename`)

image

The table editor rows are all localized: (reuses `edit`, `delete`, `deletingWithEllipsis`, `save`)

image

The confirm import dialog is localized (existing localization, somehow):

image

The key-value pairs table view is localized: (reuses `edit`, `save`, `delete`, `saving`)

image

Automated testing to ensure that the strings are passed to rendered objects within the `AddKeyRow`, `AddTableRow`, `AddTableListRow`, `ColumnMenu`, `ConfirmDeleteButton`, `ConfirmImportButton`, `DataTable`, `DataTableView`, `EditKeyRow`, `EditTableListRow`, `EditTableRow`, `KVPairs`, and `TableControls` elements.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/storage/dataBrowser/DataTableView.jsx [hunk 1]
```diff
@@ -122,7 +123,7 @@ class DataTableView extends React.Component {
               onClick={() => onViewChange(DataView.OVERVIEW)}
             >
               <FontAwesome icon="arrow-circle-left" />
-              &nbsp;Back to data
+              &nbsp;{msg.backToData()}
```
Comment (daynew) [inline apps/src/storage/dataBrowser/DataTableView.jsx:128]: Could you do quick manual check that this looks good in RTL languages? I'm worried the spacing might be off. 

Also, does the 'arrow-circle-left' look correct as well?

**Review by daynew — APPROVED** [review]
Looks good. Great tests!
Just one minor concern

**Review by smusoke — APPROVED** [review]
LGTM

**Comment by wilkie** [comment]
Fixing up styling for right-to-left languages and discovered that the data library is hidden from view:

image

Therefore, I have added the correct style to fix this. (adding a `left: 0` to the library pane regardless of direction... it doesn't need to be on one side or the other, logically)

But, either way, I have added the appropriate styles for the existing view:

image

image

**NOTE**: The markdown for the Arabic localization seems to be messed up with an extra space, so the links are very broken.

**NOTE**: The right-to-left language support (and language localization support in general) for the library module listing is dubious. It constructs '5 tables', etc, programmably and this is likely very bogus for most languages, but especially right-to-left ones as it renders horribly (But I'm not fixing that, here... we should localize this table separately as it appears in the levelbuilder as well):

image

**Review by annaxuphoto — APPROVED** [review]

File: apps/src/storage/dataBrowser/DataTableView.jsx (hunk 1 above)
Comment (wilkie) [reply apps/src/storage/dataBrowser/DataTableView.jsx:128]: I believe I have resolved this.

File: apps/src/storage/dataBrowser/DataTableView.jsx [hunk 2]
```diff
@@ -22,6 +23,7 @@ const INITIAL_STATE = {
 class DataTableView extends React.Component {
   static propTypes = {
     // from redux state
+    isRtl: PropTypes.bool.isRequired,
```
Comment (smusoke) [inline apps/src/storage/dataBrowser/DataTableView.jsx:26]: Out of curiosity is the `isRtl` pattern an existing one we already implement elsewhere to handle RTL languages?

**Review by mgc1194 — APPROVED** [review]

File: apps/test/unit/storage/dataBrowser/AddTableListRowTest.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {expect} from '../../../util/reconfiguredChai';
+import commonI18n from '@cdo/locale';
+import sinon from 'sinon';
+
+import AddTableListRow from '@cdo/apps/storage/dataBrowser/AddTableListRow';
+
+describe('AddTableListRow', () => {
```
Comment (smusoke) [inline apps/test/unit/storage/dataBrowser/AddTableListRowTest.jsx:9]: ✨ ✨ Thanks for all of the new tests ✨ ✨

**Review by smusoke — APPROVED** [review]

File: apps/src/storage/dataBrowser/DataTableView.jsx (hunk 2 above)
Comment (daynew) [reply apps/src/storage/dataBrowser/DataTableView.jsx:26]: yeah, it's all over the place unfortunately.

**Review by daynew — APPROVED** [review]

**Pull request merged**